modules/home-manager/firefox.nix

```
@@ -0,0 +1,66 @@
+{ catppuccinLib }:
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (config.catppuccin) sources;
+
+  mkFirefoxModule =
+    name:
+    let
+      cfg = config.catppuccin.${name};
+
+      # ID for the chosed theme's addon
+      addonId = sources.firefox.addonIds.${cfg.flavor}.${cfg.accent};
+      upperName = catppuccinLib.mkUpper name;
+    in
+    {
+      options.catppuccin.${name} =
+        catppuccinLib.mkCatppuccinOption {
+          inherit name;
+          accentSupport = true;
+        }
+        // {
+          finalPackage = lib.mkOption {
+            type = lib.types.package;
+            readOnly = true;
```

perhpas we should mark this as `visible = false` such that the search won't have this come up?
```suggestion
            readOnly = true;
            visible = false;
```

---

Done, although I added this option mainly so that users could still override the firefox-based browsers' wrappers' arguments.